app/src/main/java/com/willowtree/vocable/customviews/ActionButton.kt

```diff
@@ -22,8 +22,10 @@ open class ActionButton @JvmOverloads constructor(
 
     override fun sayText(text: CharSequence?) {
         text?.let {
-            VocableTextToSpeech.getTextToSpeech()
-                ?.speak(it, TextToSpeech.QUEUE_FLUSH, null, id.toString())
+            if (it.isNotBlank()) {
```

Comment:
Could potentially get rid of the `.let` and condense this to `if (text?.isNotBlank() == true)` which would cover both your false and null cases